url2df0f7c811_przemekgalazka_legacycode_tgz-pJ8-pl_cc_core_cmd_CommandJ8/src/ccproxy/AgentConnectionPing.java

```diff
@@ -0,0 +1,47 @@
+package ccproxy;
+
+import org.apache.log4j.Logger;
+
+public class AgentConnectionPing extends Thread {
+	static Logger log = Logger.getLogger(AgentConnectionPing.class);
+	
+	AgentConnection ac;
+	Thread pingThread = null, mainThread;
+	private boolean stop = false;
+	
+	public AgentConnectionPing(AgentConnection agentConnection,
+			Thread mainThread) {
+		ac = agentConnection;
+
+		this.mainThread = mainThread;
+	}
+
+	public void stopThread(){
+		stop=true;
+	}
+	
+	public void run() {
+		setName("Ping: "+ac.agentNumber);
+		pingThread = currentThread();
+		;
+
+		while (true) {
+			if (isInterrupted())
+				break;
+			if (ac.loggedout)
+				break;
+			if (stop){
+				break;
+			}
+			if (!ac.checkPing(pingThread, mainThread))
+				break;
+			try {
+				sleep(5000);
+			} catch (InterruptedException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)